include/grpc/grpc_security.h

```diff
@@ -143,6 +143,16 @@ grpc_channel_credentials *grpc_google_default_credentials_create(void);
 #define GRPC_DEFAULT_SSL_ROOTS_FILE_PATH_ENV_VAR \
   "GRPC_DEFAULT_SSL_ROOTS_FILE_PATH"
 
+/* Overrides the default path for TLS/SSL roots.
+   The path must point to a PEM encoded file with all the roots such as the one
+   that can be downloaded from https://pki.google.com/roots.pem.
+   This function is not thread-safe and must be called at initialization time
+   before any ssl credentials are created to have the desired side effect.
+   It also does not do any checks about the validity or contents of the path.
+   If the GRPC_DEFAULT_SSL_ROOTS_FILE_PATH environment is set, it will override
+   the roots_path specified in this function. */
+void grpc_override_ssl_default_roots_file_path(const char *roots_path);
```

Comment:
We actually don't need a function that loads a file based on given path but instead we need one that allows to set the PEM encoded string directly. The reason is that some languages will load the roots as a resource (for C# that is packaged directly in the binary) - and there won't be any physical file to load from.

Btw, early loading of the roots will mean that for all wrapped languages (not for C core nor C++), we will sacrifice ~300kB of memory by keeping the contents of roots.pem in memory (even if the user doesn't create any secure channels).
To me that seems like an ok price to pay for the added convenience.